pytorch_lightning/loops/fit_loop.py

```diff
@@ -188,7 +188,9 @@ def on_advance_start(self) -> None:
         model = self.trainer.lightning_module
 
         # reset train dataloader
-        if self.current_epoch != 0 and self.trainer._should_reload_dl_epoch:
```

Comment:
Hey @ninginthecloud. I might be wrong, but `self.current_epoch != 0` is just a small optimization to prevent re-creating the dataloader right ?